File: flutter/lib/domain/repository/flashcard_repository.dart
import 'package:flashcard/domain/entities/flashcard.dart';

abstract class FlashcardRepository {
  Future<void> addFlashcardToDeck(int deckId, Flashcard flashcard);
  Future<void> updateFlashcard(Flashcard updatedFlashcard);

  Future<void> deleteFlashcard(Flashcard flashcard);
  Future<void> deleteFlashcardById(int flashcardId);
  Future<void> deleteFlashcardsByDeckId(int deckId);

  Future<List<Flashcard>> getFlashcardsByDeckId(int deckId);
  Future<Flashcard> getFlashcardById(int flashcardId);
  Future<void> addMultipleFlashcardsToDeck(int deckId, List<Flashcard> flashcards);

  Future<List<Flashcard>> getAllFlashcards();
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/repository/FlashcardRepository.kt
package com.synergyboat.flashcardAi.domain.repository

import com.synergyboat.flashcardAi.domain.entities.Flashcard

interface FlashcardRepository {
    suspend fun createFlashcard(flashcard: Flashcard): Long
    suspend fun getFlashcardsByDeck(deckId: Long): List<Flashcard>
    suspend fun deleteFlashcard(flashcard: Flashcard)
    suspend fun updateFlashcard(flashcard: Flashcard)
    suspend fun getFlashcardById(id: Long): Flashcard?
}